syntax = "proto3";

service DataNodeService {
  rpc SendHeartbeat(Empty) returns (Status) {}
  rpc StoreBlock(BlockData) returns (Status) {}
  rpc DeleteBlock(BlockId) returns (Status) {}
  rpc SendBlock(BlockId) returns (BlockData) {}
  rpc CleanStart(Empty) returns (Status) {}
  rpc ChangeOfLeader(LeaderInfo) returns (Status) {}
}

message Empty {}

message BlockId {
  string id = 1;
}

message BlockData {
  string id = 1;
  bytes data = 2;
}

message LeaderInfo {
  string id = 1;
  string address = 2;
}

message Status {
  bool success = 1;
  string message = 2;
}
